### PR TITLE
switch to hercules ci

### PR DESCRIPTION
**Pull request by zowoq**

### CODE REVIEW

**Comment by zowoq** [comment]
@Mic92 Can I enable hercules for this repo, will it interfere with the current build/release process?

File: bors.toml [hunk 1]
```diff
@@ -1,10 +1,2 @@
...
-status = [
-  # garnix
-  "Evaluate flake.nix",
-  "package netboot-nixos-2211 [x86_64-linux]",
-  "package kexec-installer-nixos-2211 [x86_64-linux]",
-  "package kexec-installer-nixos-unstable [x86_64-linux]",
-  "check kexec-installer-unstable [x86_64-linux]",
-  "package netboot-nixos-unstable [x86_64-linux]"
-]
+status = [ ]
```
Comment (zimbatm) [inline bors.toml:2]: This list will need to include the Hercules CI checks

File: garnix.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+builds:
+  exclude:
+    - "*"
```
Comment (zimbatm) [inline garnix.yaml:3]: Might as well remove the Garnix app from the repo instead

Comment (zowoq) [reply garnix.yaml:3]: This is just to save some cycles, I'll drop this and the garnix app can be disabled when we're ready to merge.

**Comment by Mic92** [comment]
> @Mic92 Can I enable hercules for this repo, will it interfere with the current build/release process?

Sure.

**Comment by Mic92** [comment]
@zowoq Just so you are aware, there are people relying on these assets to be available: https://github.com/nix-community/nixos-images/releases/tag/nixos-unstable so please make sure they don't get accidentally deleted or if they do they are quickly restored.

**Comment by zowoq** [comment]
It's a bit awkward running the script in an effect so I'll leave that for now. 

I'll swap out garnix for hercules and keep the release process in the action.

**Comment by zowoq** [comment]
I'll pick this up again in the next couple of weeks.

I'll probably rewrite the script as an effect and let hercules create/merge the flake updates to avoid issues with bors, effects have some new features for publishing releases that may be useful as well.

I'll create a fork in this org to work on so I can test publishing releases without disrupting this repo.

**Comment by Kranzes** [comment]
I have some WIP fork for this, it requires a couple of features to be added to HCI's github releases effect.
https://github.com/Kranzes/nixos-images/
https://github.com/hercules-ci/hercules-ci-effects/issues/132

**Comment by zowoq** [comment]
I was looking at using the same tag (e.g. `artifacts`) for both stable and unstable and versioning the files, thought that might be simpler?

```
gh release delete artifacts
gh release create artifacts --title "build $date" "$files"
```
```
bzImage-aarch64-22.11-linux
bzImage-aarch64-unstable-linux
bzImage-x86_64-22.11-linux
bzImage-x86_64-unstable-linux
initrd-aarch64-22.11-linux
initrd-aarch64-unstable-linux
initrd-x86_64-22.11-linux
initrd-x86_64-unstable-linux
netboot-aarch64-22.11-linux.ipxe
netboot-aarch64-unstable-linux.ipxe
netboot-x86_64-22.11-linux.ipxe
netboot-x86_64-unstable-linux.ipxe
nixos-kexec-installer-aarch64-22.11-linux.tar.gz 
nixos-kexec-installer-aarch64-unstable-linux.tar.gz 
nixos-kexec-installer-x86_64-22.11-linux.tar.gz 
nixos-kexec-installer-x86_64-unstable-linux.tar.gz 
```

**Comment by Mic92** [comment]
> I was looking at using the same tag (e.g. `artifacts`) for both stable and unstable and versioning the files, thought that might be simpler?
> 
> ```
> gh release delete artifacts
> gh release create artifacts --title "build $date" "$files"
> ```
> 
> ```
> bzImage-aarch64-22.11-linux
> bzImage-aarch64-unstable-linux
> bzImage-x86_64-22.11-linux
> bzImage-x86_64-unstable-linux
> initrd-aarch64-22.11-linux
> initrd-aarch64-unstable-linux
> initrd-x86_64-22.11-linux
> initrd-x86_64-unstable-linux
> netboot-aarch64-22.11-linux.ipxe
> netboot-aarch64-unstable-linux.ipxe
> netboot-x86_64-22.11-linux.ipxe
> netboot-x86_64-unstable-linux.ipxe
> nixos-kexec-installer-aarch64-22.11-linux.tar.gz 
> nixos-kexec-installer-aarch64-unstable-linux.tar.gz 
> nixos-kexec-installer-x86_64-22.11-linux.tar.gz 
> nixos-kexec-installer-x86_64-unstable-linux.tar.gz 
> ```

I think I prefer using tags per release. Reason is that I don't want to delete old releases but stop updating them at some point. Also over time we might add/remove images.

**Comment by Kranzes** [comment]
I agree that seperate releases is the way to go, someone just needs to implement the missing features to the HCI effect.

**Comment by zowoq** [comment]
@Kranzes Do you want to take this over?

**Pull request closed without merging**